Tests/MongoSwiftTests/MongoConnectionStringTests.swift

```diff
@@ -595,6 +595,7 @@ final class ConnectionStringTests: MongoSwiftTestCase {
         expect(try connStr.applyOptions(opts)).to(throwError(errorType: MongoError.InvalidArgumentError.self))
     }
 
+#if compiler(>=5.4)
```

Comment:
I think your suspicion around this being caused by lack of shadowing support sounds right. since the changes to make the test compile should be minimal (use a unique variable name each time instead of reusing `host`) I'd suggest we just do that instead rather than conditionally defining the test.